### ISSUE TITLE
Component doesn't re-render when using with react-data-table-component

### ISSUE THREAD

**Issue by kruyvanna** [mobxjs/mobx-react-lite#290]
## Intended outcome
The data-table should re-render after state change

## Actual outcome
The data-table doesn't re-render

## How to reproduce the issue
Please check this sanbox: https://codesandbox.io/s/sad-haslett-mq9lh?file=/src/App.js

## Versions
mobx-react-lite: 2.0.7

**Comment by FredyC** [comment]
I am not aware of better option than this. Obviously, MobX cannot see into every function that's being executed insider 3rd party library.

```js
format: row => <Observer>{() => (row.done ? "Done" : "Pending")}</Observer>
```

**Comment by FredyC** [comment]
Btw, you can even optimize here and instead of sticking `observer` on a whole table, use Observer on specific columns so it doesn't have re-render a whole thing.

**Comment by kruyvanna** [comment]
It works! Thanks @FredyC

**Comment by FreeClimb** [comment]
I support the idea of observables since the knockout and mvvm template.
Аnd now in a large project I tried to use mobx-react-lite with Ant Design as widgets. I have the same problem, the Table component is not updated after any changes in the dataSource. I tried changing useObservable to observable, but the problem persists. Here is my example on codesandbox:

Ant Design Table with CRUD in Modal form

I would appreciate any help that I am doing wrong.
thank
